### DISCUSSION TITLE
Deletion of Multiple Characters

### DISCUSSION THREAD

**Discussion by Syntonica** [helix-editor/helix#9068] [Q&A] [answered]
I was fiddling about with some code and discovered I needed to delete  3 chars in the middle of a word. So, I typed `3x`. Needless to say, that didn't go over so well.  I can't find a 2-keystroke solution to doing this.  I've only found `vlld` & `ddd`.

What am I missing?

**Comment by Syntonica** [comment]
Hm, no replies.  Can I assume then that I am indeed not missing anything and that I am just bumping up against the differences in philosophies of helix and vim.

**Reply by kirawi** [reply, marked as answer]
Closest is `v<count>`+`hjkl` + `d`
It might make sense to have `d` accept a count.

Edit: I'm actually not sure it make sense upon more thought. `d` deletes a selection so manipulating the selection would be the way forward. This might be something better for plugins (unless the maintainers think otherwise).

**Reply by Syntonica** [reply]
No worries. I think I just hit an edge case with not being able to use `x` and it broke my brain.  I'll just stick with `ddd`. It may be an extra key to press, but it's still very fast.

**Reply by xxxserxxx** [reply]
It's not an edge case; I encounter this all the time. Sometimes it's wanting to move two or three characters to the beginning of the word, or wanting to delete 3 characters at the start -- it comes up a lot in file renames in `nnn` because I use Helix as the batch rename editor. I frequently have a bunch of files with generated file names, and I want to get rid of, or move, X characters somewhere in one or more files. I can `3c` and `3l`, but not `3d` and even after not having used vim for two or so years, it _still_ bites me at least once a week.

FWIW, when I remember, I usually use `%/<chars>RETd` if <chars> are all the same, or I can compose some regex that catches them all, but it's more keystrokes (and often a lot more) and far more cognitive load than being able to `3d` would be.
